integrations/clickup/main.py

```diff
@@ -0,0 +1,73 @@
+from loguru import logger
+from port_ocean.context.ocean import ocean
+from port_ocean.core.ocean_types import RAW_RESULT, ASYNC_GENERATOR_RESYNC_TYPE
+from client import ClickUpClient, ObjectKind
+from typing import Any
+
+
+def init_client() -> ClickUpClient:
+    """Initialize ClickUp client with the integration token."""
+    return ClickUpClient(
+        clickup_token=ocean.integration_config["clickup_token"],
+        host=ocean.integration_config["clickup_host"],
+    )
+
+
+@ocean.on_resync(ObjectKind.TEAM)
+async def on_resync_teams(kind: str) -> RAW_RESULT:
+    client = init_client()
+    teams = await client.get_teams()
+    logger.info(f"Received {len(teams)} teams from clickUp")
+    return teams
+
+
+@ocean.on_resync(ObjectKind.PROJECT)
+async def on_resync_projects(kind: str) -> RAW_RESULT:
+    client = init_client()
+    projects = await client.get_spaces()
+    logger.info(f"Received {len(projects)} projects")
+    return projects
+
+
+@ocean.on_resync(ObjectKind.ISSUE)
+async def on_resync_issues(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
+    client = init_client()
+    async for issues in client.get_paginated_tasks():
+        logger.info(f"Received task batch with {len(issues)} tasks")
+        yield issues
+
+
+@ocean.router.post("/webhook")
+async def handle_webhook(data: dict[str, Any]) -> dict[str, Any]:
+    client = init_client()
+    logger.warning(f"Received {len(data)} webhooks")
+    event_handlers = {
+        "task": (ObjectKind.ISSUE, client.get_task),
+        "team": (ObjectKind.TEAM, client.get_team),
+        "space": (ObjectKind.PROJECT, client.get_space),
```

Comment:
space or project? I am confused, I am not sure it will work

Reply:
you are right. This wouldn't work, I have removed the enums and used the actual names.
Kindly review and let me know if there's more I can do to improve the code. always happy to optimize and learn